lua/starfall/libs_sv/cfc_ply.lua

```diff
@@ -26,139 +26,143 @@ local function noAccess( message )
     return SF.Throw( message or "You don't have access to that function!" )
 end
 
-local function checkPlyCorePerms( ply )
-    if not ply:IsAdmin() then return noAccess() end
+local function checkPlyCorePerms( sfInstance, target )
+    local ply = sfInstance.player
+    if ply:IsAdmin() then return end
+
+    local allowed, message = hook.Run( "CFC_SFLib_PlyCoreFunction" )
+    if allowed == false then noAccess( message ) end
```

Comment:
Two things:

1. We need to pass the `sfInstance`, `ply`, and `target` into the `hook.Run`
2. For this hook to be most useful, I think it should include the name of the function being called, and any args that are passed to it. This would allow an addon to, for example, say "you can set your HP in pvp mode but only up to 50, and only once per minute" or something along those lines.

So you could do:
```suggestion
local function checkPlyCorePerms( sfInstance, target, funcName, ... )
    local ply = sfInstance.player
    if ply:IsAdmin() then return end

    local allowed, message = hook.Run( "CFC_SFLib_PlyCoreFunction", funcName, sfInstance, ply, target, ... )
    if allowed == false then noAccess( message ) end
```

And then later you could do:
```lua
-- in setHealth for example:
checkPlyCorePerms( instance, ply, "setHealth", health )
```

(You can decide the order of arguments passed into `hook.Run` - this is what made the most sense to me, but I'll leave it up to you)